home/tests/test_display_question_feature.py

```diff
@@ -107,10 +91,97 @@ def test_response_context(self, response):
             for check, expected in expectedPairs:
                 assert str(response.context[check]) == expected
 
-        @pytest.mark.parametrize(('filterType, expected'), [('date', 'Answer B'), ('votes', 'Popular Answer')])
+        @pytest.mark.parametrize(('filter_type, expected'), [('date', 'Answer B'), ('votes', 'Popular Answer')])
         @pytest.mark.django_db
-        def test_sorting_answers(self, client, filterType, expected):
-            url = f'/explore/question_14/?sortanswersby={filterType}'
+        def test_sorting_answers(self, client, filter_type, expected):
+            url = f'/explore/question_14/?sortanswersby={filter_type}'
             response = client.get(url)
-            answer = response.context['answers'].first()
+            answer = response.context['answers_structure'][0][0]
             assert str(answer.content) == expected
+
+    class TestLikesMechanism:
+        @pytest.mark.django_db
+        def test_logged_user_like(self, client):
+            """
+            in this test, a logged user press like on (answer A) which he disliked before
+            """
+            user = Profile.objects.all().get(user_id=1).user
+            client.force_login(user)
+            url = reverse('thumbs')
+            response = client.post(url, {'button': "like 9"})
+            response = client.get(response.url)
+            answerA = '(<Answer: Answer A>, \'bi bi-hand-thumbs-up-fill\', \'bi bi-hand-thumbs-down'
+            assert answerA in str(response.context['answers_structure'])
+
+        @pytest.mark.django_db
+        def test_logged_user_dislike(self, client):
+            """
+            in this test, a logged user press dislike on (answer B) which he liked before
+            """
+            user = Profile.objects.all().get(user_id=1).user
+            client.force_login(user)
+            url = reverse('thumbs')
+            response = client.post(url, {'button': "dislike 10"})
```

Comment:
Missing assertion about the response from the post reqeuest, you seem to be blindly assuming it will redirect you